scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -129,6 +129,15 @@
   (log-weird-state compute-cluster pod-name cook-expected-state-dict k8s-actual-state-dict)
   (kill-pod api-client name cook-expected-state-dict pod))
 
+(defn kill-pod-hard
+  "Kills the pod with a grace period of 0 (a 'hard kill'). This is used when pods
+  are in the :pod/deleting state for an extended period of time, to force them to
+  actually go away. We don't do this for every kill because under normal
+  circumstances, we want to give processes a chance to properly handle SIGTERM."
+  [{:keys [api-client name]} pod-name {:keys [pod]}]
+  (log/info "In compute cluster" name ", pod" pod-name "needs to be hard-killed")
+  (kill-pod api-client name nil pod :grace-period-seconds 0))
```

Comment:
We need to propagate explain why its OK to set the cook expected state dict to nil here, versus propagating the existing dictionary. (In this case, the expected state dict is missing, so what we have is equivalent.) However, this is weird enough that we should document it, or propagate it.

Reply:
Done